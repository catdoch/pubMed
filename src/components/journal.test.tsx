import React from 'react';
import { render, screen } from '@testing-library/react';
import Journal from './journal';

const data = {
  data: {
      uid: '32809478',
      authors: [{ name: 'Khatter NJ', authtype: 'Author' }],
      pubdate: '2020 Jan',
      fulljournalname: 'full journal name',
      title: 'title of journal'
  },
};

const SUT = render(<Journal data={data} />);

test('renders Journal component', () => {
    expect(SUT).toMatchSnapshot();
});
