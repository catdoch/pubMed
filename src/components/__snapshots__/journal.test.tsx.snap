// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Journal component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="journal"
        >
          <h3
            class="journal-title"
          >
            title of journal
          </h3>
          <p
            class="journal-name"
          >
            in full journal name
          </p>
          <p
            class="journal-pubDate"
          >
            2020 Jan
          </p>
          <div
            class="journal-author"
          >
            By:
             
            <p>
              Khatter NJ
            </p>
          </div>
          <a
            class="journal-link"
            href="https://pubmed.ncbi.nlm.nih.gov/32809478"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            View journal
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="journal"
      >
        <h3
          class="journal-title"
        >
          title of journal
        </h3>
        <p
          class="journal-name"
        >
          in full journal name
        </p>
        <p
          class="journal-pubDate"
        >
          2020 Jan
        </p>
        <div
          class="journal-author"
        >
          By:
           
          <p>
            Khatter NJ
          </p>
        </div>
        <a
          class="journal-link"
          href="https://pubmed.ncbi.nlm.nih.gov/32809478"
          rel="nofollow noopener noreferrer"
          target="_blank"
        >
          View journal
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
